src/parser/core/modules/PrecastStatus.js

```diff
@@ -41,6 +45,23 @@ export default class PrecastStatus extends Module {
 					type: 'applybuff',
 				})
 
+				// Determine if this buff comes from a known action, fab a cast event
+				const statusInfo = getDataBy(STATUSES, 'id', event.ability.guid)
+				if (statusInfo) {
+					const actionInfo = getDataBy(ACTIONS, 'statusesApplied', statusInfo)
+					// Action found - push it into _combatantActions array if not already there and synthesize a cast event
+					// If action has already been synthesized (and pushed into _combatantActions array), do nothing
+					if (actionInfo && this._combatantActions.indexOf(actionInfo.id) === -1) {
+						this._combatantActions.push(actionInfo.id)
+						const castEvent = event
```

Comment:
I found a pretty big bug with the precastStatus logic in `PrecastStatus.js`. 

First, I realized that the `statusesApplied` weren't being set on the right actions - Astro "Play" isn't actually the action that grants arcana buffs, it's the arcanas themselves like `ACTIONS.THE_BALANCE`

So when I fixed that, I noticed that 
1. prepull cast events were duplicated 
2. removebuff events for the action were having the wrong guid
3. It wouldn't get picked up by the main module loop

Upon closer inspection, my suspicion is that this line here is modifying the `removebuff` event directly rather than creating a clone. I'm not exactly sure if just doing {...event} will work though so I'll keep testing and get back to you.

Reply:
Actually, this is all it took, changing the contents from this line 56-61:
```
						events.splice(0, 0, {
							...event,
							ability: {
								...event.ability,
								name: actionInfo.name,
								abilityIcon: actionInfo.icon.replace('https://xivapi.com/i/', '').replace('/', '-'),
								guid: actionInfo.id,
							},
							timestamp: startTime - 2,
							type: 'cast',
						})
					}
```